superbench/benchmarks/micro_benchmarks/cpu_copy_performance/cpu_copy.cu

```diff
@@ -0,0 +1,319 @@
+                                     {"num_loops", required_argument, nullptr, static_cast<int>(OptIdx::kNumLoops)},
+                                     {"check_data", no_argument, nullptr, static_cast<int>(OptIdx::kEnableCheckData)}};
+    int getopt_ret = 0;
+    int opt_idx = 0;
+    bool size_specified = false;
+    bool num_warm_up_specified = false;
+    bool num_loops_specified = false;
+    bool parse_err = false;
+
+    while (true) {
+        getopt_ret = getopt_long(argc, argv, "", options, &opt_idx);
+        if (getopt_ret == -1) {
+            if (!size_specified || !num_warm_up_specified || !num_loops_specified) {
+                parse_err = true;
+            }
+            break;
+        } else if (getopt_ret == '?') {
+            parse_err = true;
+            break;
+        }
+        switch (opt_idx) {
+        case static_cast<int>(OptIdx::kSize):
+            if (1 != sscanf(optarg, "%lu", &(opts->size))) {
+                std::cerr << "Invalid size: " << optarg << std::endl;
+                parse_err = true;
+            } else {
+                size_specified = true;
+            }
+            break;
+        case static_cast<int>(OptIdx::kNumWarmUp):
+            if (1 != sscanf(optarg, "%lu", &(opts->num_warm_up))) {
+                std::cerr << "Invalid num_warm_up: " << optarg << std::endl;
+                parse_err = true;
+            } else {
+                num_warm_up_specified = true;
+            }
+            break;
+        case static_cast<int>(OptIdx::kNumLoops):
+            if (1 != sscanf(optarg, "%lu", &(opts->num_loops))) {
+                std::cerr << "Invalid num_loops: " << optarg << std::endl;
+                parse_err = true;
+            } else {
+                num_loops_specified = true;
+            }
+            break;
+        case static_cast<int>(OptIdx::kEnableCheckData):
+            opts->check_data = true;
+            break;
+        default:
+            parse_err = true;
+        }
+        if (parse_err) {
+            break;
+        }
+    }
+
+    if (parse_err) {
+        PrintUsage();
+        return -1;
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Benchmark the memory copy performance between two NUMA nodes.
+ *
+ * This function measures the performance of copying memory from a source NUMA node to a destination NUMA node.
+ *
+ * @param src_node The source NUMA node from which memory will be copied.
+ * @param dst_node The destination NUMA node to which memory will be copied.
+ * @param opts A reference to an Opts structure containing various options and configurations for the benchmark.
+ * @return The performance metric of the memory copy operation, typically in terms of bandwidth or latency.
+ */
+double BenchmarkNUMACopy(int src_node, int dst_node, Opts &opts) {
+    int ret = 0;
+
+    // Set CPU affinity to the NUMA node with CPU cores assoiated
+    int affinity_node = HasCPUsForNumaNode(src_node) ? src_node : dst_node;
```

Comment:
If we proceed with the changes to enable dst_node being a memory only numa node, we can avoid setting the affinity_node, as the src_node is guaranteed to have cpu cores due to earlier checks.

Reply:
also, can avoid rechecking HasCPUsForNumaNode as upstream checks are already validating

Reply:
> If we proceed with the changes to enable dst_node being a memory only numa node, we can avoid setting the affinity_node, as the src_node is guaranteed to have cpu cores due to earlier checks.

We also need to measure the bandwidth between a CPU node and another CPU node. Therefore, the dst_node can either be a memory-only node or a CPU node.

Reply:
Apologies, I framed my comment which gives wrong notion. The upstream checks are validating that the src_node is always guaranteed with cpus. So we can cleanup this part to use the src_node for numa affinity.
the dst can be a cpu node or a memory only numa node.

This is based on the other making changes mentioned in the other [comment ](https://github.com/microsoft/superbenchmark/pull/662#discussion_r1841172566) removing the check  for dst_node
```
if (!HasMemForNumaNode(dst_node)) {
                // Skip the NUMA node if there are no CPUs available
                continue;
            }
```